src/main/resources/META-INF/rewrite/spring-boot-24.yml

```diff
@@ -158,6 +158,7 @@ recipeList:
         - org.springframework.test.context.junit4.SpringRunner
         - org.springframework.test.context.junit4.SpringJUnit4ClassRunner
       extension: org.springframework.test.context.junit.jupiter.SpringExtension
+  - org.openrewrite.java.spring.boot2.ReplaceExtendWithAndContextConfiguration
```

Comment:
In the past we've occasionally held off on making new recipes part of the default chain of upgrades until after they had been tried on a large-ish set of projects, for instance through public.moderne.io. Just because when applied at scale, there might be uncovered edge cases that could trip up other users.

I'm impressed by the attention to detail and test cases already covered here; have you also already tried this on a few projects there? That might give us the confidence we need to indeed make this part of the default chain of recipes immediately.

Reply:
I wasn't aware of public.moderne.io and didn't try it on any projects there. I did test it locally on one of our projects. I wouldn't mind holding off on including it in a chain of upgrades. I think the important piece is that it eventually becomes part of the JUnit 4 to JUnit 5 migration - However that happens is fine with me.